verification/spec/connection-handshake/ConnectionHandshakeModule.tla

```diff
@@ -0,0 +1,294 @@
+     connectionState |-> store.connection.state]
+
+
+(* TODO *)
+GetClientProof ==
+    IF store.client # nullClient
+    THEN [content |-> "client proof content",
+          height |-> store.client.height]
+    ELSE [content |-> "client proof content",
+          height |-> nullClient]
+
+
+(***************************************************************************
+ Connection Handshake Module actions & operators.
+ ***************************************************************************)
+
+
+(* Drops a message without any priming of local variables. 
+    This action always enables (returns true); use with care.
+    This action is analogous to "abortTransaction" function from ICS specs.
+ *)
+DropMsg(m) ==
+    /\ UNCHANGED<<outBuf, store>>
+
+
+
+(* Modifies the local store, by replacing the connection with the input 'newCon'.
+    If the height of the chains has not yet attained MaxHeight, this action also
+    advances the chain height. 
+ *)
+ModifyStore(newCon) ==
+    \/ /\ store.height < MaxHeight
+       /\ store' = [store EXCEPT !.connection = newCon,
+                                 !.height = @ + 1]
+    \/ /\ store.height >= MaxHeight
+       /\ store' = [store EXCEPT !.connection = newCon]
+
+
+(* State predicate, guarding the handler for the Init msg. 
+ *)
+PreconditionsInitMsg(m) ==
+    /\ store.connection.state = "UNINIT"
+    /\ Len(outBuf) < MaxBufLen  (* Enables if we can reply on the buffer. *)
+
+
+(* Handles a "CHMsgInit" message 'm'.
+    
+    Primes the store.connection to become initialized with the parameters
+    specified in 'm'. Also creates a reply message, enqueued on the outgoing
+    buffer.
+ *)
+HandleInitMsg(m) ==
+    (* The good-case path. *)
+    \/ /\ PreconditionsInitMsg(m) = TRUE
+       /\ LET newCon == [parameters |-> m.parameters, 
+                         state      |-> "INIT"]
+              sProof == GetStateProof
+              cProof == GetClientProof
+              replyMsg == [parameters |-> FlipConnectionParameters(m.parameters),
+                           type |-> "CHMsgTry",
+                           stateProof |-> sProof,
+                           clientProof |-> cProof]
+           IN /\ outBuf' = Append(outBuf, replyMsg)
+              /\ ModifyStore(newCon)
+    (* The exceptional path. *)
+    \/ /\ PreconditionsInitMsg(m) = FALSE
+       /\ DropMsg(m)
+
+
+(* State predicate, guarding the handler for the Try msg. 
+ *)
+PreconditionsTryMsg(m) ==
+    /\ \/ store.connection.state = "UNINIT"
+       \/ store.connection.state = "INIT" /\ CheckLocalParameters(m.parameters)
+    /\ Len(outBuf) < MaxBufLen
+
+
+(* Handles a "CHMsgTry" message.
+ *)
+HandleTryMsg(m) ==
+    \/ /\ PreconditionsTryMsg(m) = TRUE
+       /\ LET newCon == [parameters |-> m.parameters, 
+                         state |-> "INIT"]
+              sProof == GetStateProof
+              cProof == GetClientProof
+              replyMsg == [parameters |-> FlipConnectionParameters(m.parameters),
```

Comment:
Looks like the proof related spec is not finished? Here we need to check the proofs from message `m`. also include local proofs in replyMsg.